react/components/ItemContainer.js

```diff
@@ -1,156 +1,35 @@
-      } db pv3 no-underline outline-0 tl link t-small truncate c-muted-1 underline-hover`,
-      {
-        pr5: menuPosition === categoryMenuPosition.DISPLAY_LEFT.value,
-        pl5: menuPosition === categoryMenuPosition.DISPLAY_RIGHT.value,
-        ph5: menuPosition === categoryMenuPosition.DISPLAY_CENTER.value,
-      }
-    )
-
-    if (parentSlug) params.subcategory = subItem.slug
-    return (
-      <li key={subItem.id} className="list pa0">
-        <Link
-          onClick={onCloseMenu}
-          page={
-            parentSlug ? 'store.search#subcategory' : 'store.search#category'
-          }
-          className={secondLevelLinkClasses}
-          params={params}
-        >
-          {subItem.name}
-        </Link>
-      </li>
-    )
-  }
-
-  shouldRenderSecondLevel(category) {
-    const { children } = category
-    const { showSecondLevel } = this.props
-
-    return children && children.length > 0 && showSecondLevel
-  }
-
-  renderChildren(category) {
-    const { parentSlug } = this.props
-    return (
-      this.shouldRenderSecondLevel(category) &&
-      category.children.map(subCategory =>
-        this.renderLinkSecondLevel(parentSlug, category, subCategory)
-      )
-    )
-  }
-
-  render() {
-    const { containerStyle, categories, parentSlug, menuPosition } = this.props
-
-    const containerClasses = classNames('w-100 flex flex-wrap pa0 list mw9', {
-      'justify-start': menuPosition === categoryMenuPosition.DISPLAY_LEFT.value,
-      'justify-end': menuPosition === categoryMenuPosition.DISPLAY_RIGHT.value,
-      'justify-center':
-        menuPosition === categoryMenuPosition.DISPLAY_CENTER.value,
-    })
-
-    const columnItemClasses = classNames({
-      'pl0 pr7': menuPosition === categoryMenuPosition.DISPLAY_LEFT.value,
-      'pr0 pl7': menuPosition === categoryMenuPosition.DISPLAY_RIGHT.value,
-    })
-
-    return (
-      <div
-        className={`${
-          categoryMenu.itemContainer
-        } absolute w-100 left-0 bg-base pb2 bw1 bb b--muted-3`}
-        style={containerStyle}
-      >
-        <Container className="justify-center w-100 flex">
-          <ul className={containerClasses}>
-            {categories.map(category => (
-              <li key={category.id} className="dib">
-                <ul className={columnItemClasses}>
-                  {this.renderLinkFirstLevel(parentSlug, category)}
-                  {this.renderChildren(category)}
-                </ul>
-              </li>
-            ))}
-          </ul>
-        </Container>
-      </div>
-    )
-  }
-}
+import React from 'react'
+import { useState, useRef } from 'react'
+import PropTypes from 'prop-types'
+
+import categoryMenu from '../categoryMenu.css'
+
+const ItemContainer = ({ children, itemKey }) => {
+  const [isHovered, setIsHovered] = useState(false)
+  const containerRef = useRef(null)
+
+  return (
+    <li
+      className={`${categoryMenu.itemContainer} flex items-center db list`}
+      key={itemKey}

```

Comment:
This `key` doesn't work. Keys should be at the top level of a `map`.

Check this:
https://reactjs.org/docs/lists-and-keys.html#extracting-components-with-keys